coorx/coordinates.py

```diff
@@ -168,9 +222,284 @@ def coordinates(self):
+        self._system = p1.system
+        # Cache displacement? For now, compute on demand.
+        # self._displacement = self._p2.coordinates - self._p1.coordinates
+
+    @property
+    def p1(self) -> PointArray:
+        """The starting points of the vectors."""
+        return self._p1
+
+    @property
+    def p2(self) -> PointArray:
+        """The ending points of the vectors."""
+        return self._p2
+
+    @property
+    def system(self) -> CoordinateSystem:
+        """The coordinate system of the vector endpoints."""
+        return self._system
+
+    @property
+    def displacement(self) -> np.ndarray:
+        """The displacement array (p2.coordinates - p1.coordinates)."""
+        # Ensure coordinates are used for subtraction
+        return self._p2.coordinates - self._p1.coordinates
+
+    @property
+    def shape(self) -> tuple:
+        """The shape of the array structure (like PointArray)."""
+        return self.displacement.shape
+
+    @property
+    def ndim(self) -> int:
+        """The number of dimensions of the array structure (like PointArray)."""
+        return self.displacement.ndim
+
+    @property
+    def dtype(self) -> np.dtype:
+        """The data type of the vector components."""
+        # Displacement dtype might differ from p1/p2 if they are integers
+        return self.displacement.dtype
+
+    def __len__(self) -> int:
+        """The number of vectors in the array (first dimension, like PointArray)."""
+        return len(self._p1)
+
+    def __getitem__(self, index) -> np.ndarray:
+        """Get the displacement vector(s) at the given index."""
+        # Return the displacement component, not a new VectorArray slice
+        return self.displacement[index]
+
+    def __iter__(self):
+        """Iterate over the displacement vectors."""
+        yield from self.displacement
+
+    def _check_vector_operand(self, other: VectorArray):
+        """Validate operand for vector-vector operations."""
+        if not isinstance(other, (Vector, VectorArray)):
+            raise TypeError(f"Operand must be a VectorArray or Vector (received {type(other)})")
+        if self.system is not other.system:
+            raise ValueError(
+                f"Operand system '{other.system}' does not match this VectorArray's system '{self.system}'"
+            )
+        # Shape check for addition requires broadcast compatibility or equality
+        # Simple equality check for now, consider broadcasting later if needed.
+        if self.shape[-1] != other.shape[-1]:
+            raise ValueError(f"Operand shapes {self.shape} and {other.shape} are not compatible.")
+
+    def __add__(self, other: VectorArray | PointArray) -> VectorArray | PointArray:
+        """
+        Add this vector array to another vector array or a point array.
+
+        Vector + Vector = Vector (Result starts at self.p1)
+        Vector + Point = Point
+        """
+        if isinstance(other, (Vector, VectorArray)):
```

Comment:
No need to manually calculate displacement - just make a new vector: 
v1 = p2 - p1
v2 = p3 - p4
v1 + v2 = (p2 + v2) - v1

Reply:
Or v1 + v2 = Vector(p1, p2 + v2) to skip some operand checking